### ISSUE TITLE
Emit "scriptLoaded" event

### ISSUE THREAD

**Issue by weinand** [Microsoft/vscode-node-debug2#118]
To finish the ["Loaded Scripts Explorer"](https://github.com/Microsoft/vscode/issues/28521) I'd need a custom `scriptLoaded` event from "node-debug2":

Here is the code from node-debug:
```js
this._node.on('afterCompile', (event: NodeV8Event) => {
    this.sendEvent(new Event('scriptLoaded', { path: this._scriptToPath(event.body.script) }));
});
```

The "path" attribute should use the same form as the paths used in the `getLoadedScripts` request.

Tomorrows insider should have the "Loaded Scripts Explorer" (and if not VS Code will ignore the event).

(Since I've noticed that the result format of the `getLoadedScripts` request is utterly redundant, I'm planning to simplify this today. But I will create a separate issue for this and I will continue to support the old and new format in parallel..).

**Comment by roblourens** [comment]
I'm already sending an event of type 'script' for VS, I guess these are both private/experimental extensions to the protocol. Hopefully we can consolidate them in the future, per https://github.com/Microsoft/vscode-debugadapter-node/issues/108

**Comment by roblourens** [comment]
I can't get this to work. I'm sending scriptLoaded events but usually only the `<node_internals>` scripts show up. Sometimes everything shows up, but only about 1/10 times. It seems like it's also calling getLoadedScripts initially, and those scripts show up, but that's called before my `.../app.js` scripts are loaded.

**Comment by weinand** [comment]
@roblourens you are probably running into Sandeep's bad fix for https://github.com/Microsoft/vscode/issues/31191
Please try again with today's Insiders.

**Comment by weinand** [comment]
With a good fix for https://github.com/Microsoft/vscode/issues/31191 and a [node2 specific event massaging](https://github.com/Microsoft/vscode-node-debug/blob/master/src/node/extension/loadedScripts.ts#L42-L47), the loaded scripts show and update correctly for the "inspector" protocol.

**Comment by roblourens** [comment]
I'll send a scriptLoaded event so I can leave the VS event alone, and send the right path for vscode. It's easy and I have the change ready anyway :)

**Comment by weinand** [comment]
even better! 
BTW, you can check the `InitializeRequest.clientId if you only want to emit custom events if the corresponding client listens:
see https://github.com/Microsoft/vscode-debugadapter-node/wiki/Client-IDs

**Comment by roblourens** [comment]
There can be lots of eval scripts:

![image](https://user-images.githubusercontent.com/323878/28629209-ad28d8fc-71db-11e7-8a9d-4e0fe8e9c430.png)

What if vscode-node-debug puts them under a root item called `<eval>`, or `"Dynamic scripts"`? Or, should the adapter change the name of these and add `<eval>/`?

**Comment by weinand** [comment]
"legacy" puts the unnamed scripts under `<node_internals>` so they don't clutter the top level:

![2017-07-26_17-41-40](https://user-images.githubusercontent.com/1898161/28630240-084e2374-722a-11e7-932c-82ed05afd20e.png)

But we can consider to introduce another "well known" category: I like the `<eval>`.

If you like you try this, you can do this already in node-debug2 without touching "loadedScripts.ts". Just return the paths for eval scripts with the `<eval>` prefix and the tree will show them correctly (only sorting will be off because "loadedScripts.ts" treats `<node_internals>` in a special way, but we could generalise this to sort all `<*>` at the end). 
And you'll have to recognise all incoming paths starting with `<eval>` as well...

**Comment by roblourens** [comment]
Implemented in https://github.com/Microsoft/vscode-chrome-debug-core/commit/e086511eedf34e38c0262b3c553535d8e9ddaf41 and  https://github.com/Microsoft/vscode-chrome-debug-core/commit/c1b54be909afc686aae3ea5788d8b5402743bb86
